### PR TITLE
Remove Vim hack, adding vim8027

### PR DESCRIPTION
**Pull request by blueyed**
This is WIP to require a specific Vim version, where async is fixed.

### CODE REVIEW

**Comment by blueyed** [comment]
8.0.0027 is green (yay!), but vim-master (same version) failed: https://travis-ci.org/neomake/neomake/jobs/166551569 - restarting.

**Comment by blueyed** [comment]
The restart worked (https://travis-ci.org/neomake/neomake/jobs/166551569), so it is still flaky.. :/
Need to investigate.

**Comment by blueyed** [comment]
Seems this can happen when a job gets canceled too fast:
```
    (8/9) [EXECUTE] neomake#CancelJob
      > Neomake [2] [1.078]: Starting: /bin/zsh -c sh -c 'while true; do sleep 0.1; done'
      > Neomake [3] [1.079]: Stopping job: 19
    (8/9) [EXECUTE] (X) Jobs did not finish after 5s.
      > function <SNR>139_wait_for_finished_jobs, line 6
    (9/9) [ BEFORE] (X) Jobs did not finish after 5s.
    > function <SNR>139_wait_for_finished_jobs, line 6
```

`:h job_stop` says:

> 		The status of the job isn't checked, the operation will even
		be done when Vim thinks the job isn't running.

Filed https://github.com/vim/vim/issues/1155.

**Comment by blueyed** [comment]
@saaguero 
What do you think?
We could have a variable to use it also with older Vim versions (basically then the previous patch level you required).

File: tests/integration.vader [hunk 1]
```diff
@@ -250,3 +250,35 @@ Execute (Neomake with windo):
...
+      \ }
+  let g:neomake_maker2_maker = {
+      \ 'name': '2',
+      \ 'exe': 'sh',
+      \ 'args': ['-c', 'echo 2'],
+      \ 'remove_invalid_entries': 0,
+      \ 'errorformat': '%m',
+      \ }
+  let b:neomake_enabled_makers = ['maker1', 'maker2']
+  Neomake! maker1 maker2
```
Comment (blueyed) [inline tests/integration.vader]: Maybe it needs something slower to startup?!  (like the ruby stuff)
Or there is "just" a bug on Windows here?!

**Comment by blueyed** [comment]
For reference:

```
( 5/18) [EXECUTE] Neomake: handle result for current window
  > [verb ] [0.051]: Starting: sh -c sleep 0.1; echo finished /testplugin/[Vader-workbench]
  > [debug] [0.052]: [#3] Vim job: {'status': 'run', 'stoponexit': 'term', 'exitval': 0, 'exit_cb': 0, 'channel': channel 3 open, 'process': 11}
  > [debug] [0.052]: [#3] Vim channel: {'status': 'open', 'id': 3, 'in_io': 'pipe', 'err_mode': 'RAW', 'in_status': 'open', 'out_io': 'pipe', 'err_io': 'pipe', 'err_status': 'open', 'out_timeout': 2000, 'out_status': 'open', 'out_mode': 'RAW', 'in_mode': 'RAW', 'in_timeout': 2000, 'err_timeout': 2000}
  > [debug] [0.063]: MakeHandlerVim: stdout: channel 2 closed
  > [quiet] [0.064]: neomake#MakeHandler: stdout: job not found: 2
( 5/18) [EXECUTE] (X) Vim(let):E916: not a valid job
  > function <SNR>30_wait_for_finished_jobs[4]..neomake#MakeHandlerVimClose, line 1
```

(from https://travis-ci.org/neomake/neomake/jobs/166920717)

**Comment by blueyed** [comment]
Regarding job IDs in general, Neomake should/could its own counter, and not start with 0 just because Vim does so (ref: https://github.com/neomake/neomake/pull/706).

**Comment by bounceme** [comment]
seems like vim is buggier than neovim with async stuff, it is newer though.

**Comment by bounceme** [comment]
my live checking works with vim surprisingly haha
https://gist.github.com/bounceme/a2cb5c341af1b33c48689f64c678235f

**Comment by blueyed** [comment]
> my live checking works with vim surprisingly haha

Which makers and which Vim version are you using?  On Neomake master, or this branch?

**Comment by bounceme** [comment]
master - jshint and filetype javascript - vim 8.32 neovim 1.5

**Comment by blueyed** [comment]
Rebased and cleaned up.

@BimbaLaszlo
Please try if it still works for you.

**Comment by BimbaLaszlo** [comment]
Work well. :+1:

**Comment by blueyed** [comment]
@BimbaLaszlo 
Thanks for your help! :)

**Comment by BimbaLaszlo** [comment]
Thanks for the plugin! :wink:

**Pull request merged**